Examples/MAX32660/SPIMSS_DMA/main.c

```diff
@@ -0,0 +1,216 @@
+* count.(BitCount/8) is defined as transfer unit size.
+* tx_data and rx_data is define as uint16_t. So, unit
+* size is 16/8 = 2.
+*/
+#define DATA_VALUE 0x0102
+#define TEST_COUNT 100 // Transaction count.
+
+/**
+ * Transfer unit size is 2. So, we define tx_data and rx_data as uint16_t.
+*/
+uint16_t tx_data[NUMBER_OF_UNIT_SIZE] = { 0 };
+uint16_t rx_data[NUMBER_OF_UNIT_SIZE] = { 0 };
+mxc_spimss_req_t spimss_dma_req;
+
+static uint8_t dma_tx_completed = 0;
+static uint8_t dma_rx_completed = 0;
+
+#if !SPIMSS_AUTO_DMA_HANDLING
+int txChannel = -1;
+int rxChannel = -1;
+#endif
+
+void spimss_callback(mxc_spimss_req_t *req, int error_code)
+{
+    if (NUMBER_OF_UNIT_SIZE * TRANSFER_UNIT_SIZE == req->rx_num) {
+#if SPIMSS_AUTO_DMA_HANDLING
+        dma_rx_completed = 1;
+#else
+        dma_rx_completed = 1;
+        MXC_DMA_DisableInt(rxChannel);
+        MXC_DMA_SetChannelInterruptEn(rxChannel, false, false);
+        MXC_DMA_ReleaseChannel(rxChannel);
+        MXC_SPIMSS_SetRXDMAChannel(SPIMSS, -1);
+        SPIMSS->dma &= ~MXC_F_SPIMSS_DMA_RX_DMA_EN; // Disable SPIMSS RX DMA requests
+#endif
+    } else if (NUMBER_OF_UNIT_SIZE * TRANSFER_UNIT_SIZE == req->tx_num) {
+#if SPIMSS_AUTO_DMA_HANDLING
+        dma_tx_completed = 1;
+#else
+        //Disable interrupts.
+        dma_tx_completed = 1;
+        MXC_DMA_DisableInt(txChannel);
+        MXC_DMA_SetChannelInterruptEn(txChannel, false, false); // Disable ctz interrupt.
+        MXC_DMA_ReleaseChannel(txChannel);
+        MXC_SPIMSS_SetTXDMAChannel(SPIMSS, -1);
+        MXC_SPIMSS->dma &= ~MXC_F_SPIMSS_DMA_TX_DMA_EN; // Disable SPIMSS TX DMA requests
+
+        if (req->deass) {
+            MXC_SPIMSS->mode |= MXC_F_SPIMSS_MODE_SSV; // Set Slave Select to HIGH.
+        }
+        MXC_SPIMSS->ctrl &= ~MXC_F_SPIMSS_CTRL_ENABLE; // Disable SPIMSS transaction.
+#endif
+    }
+}
+
+int main()
+{
+    int i = 0;
+
+    printf("\n\n************** SPIMSS-DMA Master Demo ****************\n");
+    printf("This example configures the SPIMSS to send data between the MISO (P0.10) and\n");
+    printf("MOSI (P0.11) pins over dma channel 0 and dma channel 1.  Connect these two pins\n");
+    printf("together. This demo shows 1024 byte data transfer for 100 times using dma.\n");
+    printf("During this demo you may see junk data printed to the serial port because the\n");
+    printf("console UART shares the same pins as the SPIMSS.One DMA channel is used as tx from\n");
+    printf("memory to SPIMSS TX FIFO and another dma channel 1 is used for reading data from\n");
+    printf("SPIMSS RX FIFO to memory.\n\n");
+
+    // Preparation of the transmission data and receive data buffer.
+    for (i = 0; i < NUMBER_OF_UNIT_SIZE; i++) {
+        tx_data[i] = DATA_VALUE + i;
+        rx_data[i] = 0;
+    }
+
+    // Initialize SPIMSS with MAP_A.
+    // For MAP_A P0.10:MISO, P0.11:MOSI, P0.12:SCK, P0.13:SSEL
+    if (0 != MXC_SPIMSS_Init(SPIMSS, 0, SPI_SPEED, MAP_A)) {
+        return -1;
```

Comment:
It's good practice to use error codes instead of hard values `0` and `-1` to make error correcting easier and have all our error code values consistent. Note, there may still be some old files in the MSDK that use hard values instead of error codes which should be corrected. You can reference the other examples to see how they do it and `mxc_errors.h` for the list of possible error codes.

In this case:
error 0: no error or success
error -1: pointer is NULL

You can do something similar to this:
```
int error;
error = MXC_SPIMSS_Init(SPIMSS, 0, SPI_SPEED, MAP_A);
if (error != E_NO_ERROR) {
    printf("Initialization error: %d\n", error);
    return error;
}
```